packages/server/tests/helpers/utils.ts

```diff
@@ -150,4 +153,39 @@ export class Utils {
     }
     return res.toString();
   };
+
+  static ethCallWRetries = async (
+    relay: RelayClient,
+    callData: { from: string; to: any; gas: string; data: string },
+    blockNumber: string,
+    requestId: string,
+  ): Promise<string> => {
+    let numberOfCalls = 0;
+    let res = await relay.call(RelayCall.ETH_ENDPOINTS.ETH_CALL, [callData, blockNumber], requestId);
+    while (res === '0x' && numberOfCalls < 3) {
+      await new Promise((r) => setTimeout(r, 2000));
+      res = await relay.call(RelayCall.ETH_ENDPOINTS.ETH_CALL, [callData, blockNumber], requestId);
+      numberOfCalls++;
+    }
+    return res;
+  };
+
+  // Handles the odd case where the contract is loading in the factory but for some reason
+  // ethers is not able to deploy it.  The ethers deploy returns a 4001 error.
+  static deployContract = async (contractFactory: ContractFactory): Promise<ethers.Contract> => {
```

Comment:
This method will indefinitely retry under the right circumstances.
You should limit the number if retries and update the method name appropriately.

Reply:
Fixed.